Library/Homebrew/reinstall.rb

```diff
@@ -0,0 +1,63 @@
+require "formula_installer"
+require "development_tools"
+require "messages"
+
+module Homebrew
+  module_function
+
+  def reinstall_formula(f, build_from_source: false)
+    if f.opt_prefix.directory?
+      keg = Keg.new(f.opt_prefix.resolved_path)
+      keg_had_linked_opt = true
+      keg_was_linked = keg.linked?
+      backup keg
+    end
+
+    build_options = BuildOptions.new(Options.create(ARGV.flags_only), f.options)
+    options = build_options.used_options
+    options |= f.build.used_options
+    options &= f.options
+
+    fi = FormulaInstaller.new(f)
+    fi.options              = options
+    fi.invalid_option_names = build_options.invalid_option_names
+    fi.build_bottle         = ARGV.build_bottle? || (!f.bottled? && f.build.bottle?)
+    fi.interactive          = ARGV.interactive?
+    fi.git                  = ARGV.git?
+    fi.link_keg           ||= keg_was_linked if keg_had_linked_opt
+    fi.build_from_source    = true if build_from_source
```

Comment:
could the right-hand side just be `build_from_source`?

Reply:
No.`FormulaInstaller` internally sets `build_from_source` from `ARGV`, e.g. as in `brew reinstall <keg> --build-from-source`. If the relinkage detects broken kegs, this line is triggered to force the reinstall from source; otherwise, the parameter is left alone.

Reply:
sounds good; thanks for the explanation